apple/REANodesManager.mm

```diff
@@ -534,6 +536,27 @@ - (NSString *)obtainProp:(nonnull NSNumber *)viewTag propName:(nonnull NSString
   } else if ([propName isEqualToString:@"zIndex"]) {
     NSInteger zIndex = view.reactZIndex;
     result = [@(zIndex) stringValue];
+  } else if ([propName isEqualToString:@"width"]) {
+    result = [@(view.frame.size.width) stringValue];
+  } else if ([propName isEqualToString:@"height"]) {
+    result = [@(view.frame.size.height) stringValue];
+  } else if ([propName isEqualToString:@"top"]) {
+    result = [@(view.frame.origin.y) stringValue];
+  } else if ([propName isEqualToString:@"left"]) {
+    result = [@(view.frame.origin.x) stringValue];
+  } else if ([propName isEqualToString:@"backgroundColor"]) {
+#if !TARGET_OS_OSX
+    UIColor *color = view.backgroundColor;
+    const size_t totalComponents = CGColorGetNumberOfComponents(color.CGColor);
+    const CGFloat *components = CGColorGetComponents(color.CGColor);
+    int r = 255 * components[MIN(0, totalComponents - 2)];
+    int g = 255 * components[MIN(1, totalComponents - 2)];
+    int b = 255 * components[MIN(2, totalComponents - 2)];
+    result = [NSString stringWithFormat:@"#%02x%02x%02x", r, g, b];
+#else
+    result = @"Cant get background color on macos";
```

Comment:
> Maybe we should come up with macOS implementation anyway?

I think it's a good idea, but I'd rather do it in separate PR. This one needs to be tested on both Paper and Fabric and both IOS and Android, which means that it is difficult to test. Its also very big.

> Maybe we should throw an exception instead?

We could, but it would crash the app, instead of reporting one of the tests as a failing, so I think I would like to keep returning a string here. I've tested it and, although it's not a pattern, it is very easy to debug. And it may be just a temporary solution, until we implement it for macOS

> The error message is poorly written.

I'll try to write it better

Reply:
> Maybe we should come up with macOS implementation anyway?

Sure!